Flaky test: TestIndicatorLightsUnknownEnergyLevel fails due to race condition with free energy checker

## Description

`TestIndicatorLightsUnknownEnergyLevel` in the energy plugin is flaky due to a race condition with the free energy checker goroutine.

## Root Cause

In `manager.go:163`, `Start()` launches `runFreeEnergyChecker()` as a goroutine:
```go
go m.runFreeEnergyChecker()
```

This goroutine immediately calls `checkFreeEnergy()` (line 745), which runs asynchronously after `Start()` returns.

The test has a race condition:
1. `Start()` returns
2. Test immediately calls `mockClient.ClearServiceCalls()`
3. Meanwhile, `runFreeEnergyChecker()` is running in a goroutine and calling `checkFreeEnergy()` which updates state and triggers `updateIndicatorLights()`
4. When the test time (05:55 UTC) is within the free energy window (21:00-07:00 UTC), this triggers a `light.turn_on` call AFTER `ClearServiceCalls()`

## Failing Test Output

```
=== RUN   TestIndicatorLightsUnknownEnergyLevel
DEBUG  energy/manager.go:831  Within free energy time (spans midnight)  {"now": "2025-12-29T05:55:09.133Z", "start": "2025-12-29T21:00:00.000Z", "end": "2025-12-29T07:00:00.000Z"}
INFO   energy/manager.go:777  Free energy availability changed  {"is_free_energy": true, "is_grid_available": true}
...
Error: "[{light turn_on map[brightness_pct:%!s(int=0) entity_id:[light.apollo_bedroom_rgb] rgb_color:[%!s(int=0) %!s(int=0) %!s(int=0)]] ...}]" should have 0 item(s), but has 1
Test: TestIndicatorLightsUnknownEnergyLevel
Messages: Expected no light.turn_on service calls for unknown energy level
--- FAIL: TestIndicatorLightsUnknownEnergyLevel (0.00s)
```

## Suggested Fix

Add a small sleep after `Start()` to allow the free energy checker goroutine to complete its initial check before clearing service calls:

```go
err := manager.Start()
assert.NoError(t, err)
defer manager.Stop()

// Wait for async goroutines to complete their initial work
time.Sleep(50 * time.Millisecond)

// Clear any service calls from startup
mockClient.ClearServiceCalls()
```

Alternatively, disable free energy checking in the test config or mock the time to be outside the free energy window.

## CI Run

https://github.com/NickBorgersOnLowSecurityNode/home-automation/actions/runs/20565979662/job/59064277756

--- a/homeautomation-go/internal/plugins/energy/manager_test.go
+++ b/homeautomation-go/internal/plugins/energy/manager_test.go
@@ -1053,6 +1053,9 @@ func TestIndicatorLightsUnknownEnergyLevel(t *testing.T) {
 	assert.NoError(t, err)
 	defer manager.Stop()
 
+	// Wait for async goroutines (e.g., runFreeEnergyChecker) to complete initial work
+	time.Sleep(50 * time.Millisecond)
+
 	// Clear any service calls from startup
 	mockClient.ClearServiceCalls()
 

